From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Fiona Ebner <f.ebner@proxmox.com>
Date: Wed, 30 Sep 2026 15:43:55 +0200
Subject: [PATCH] PVE backup: vma writer: add config: set error when allocation
 fails

Otherwise, the information about which operation failed is lost.

Signed-off-by: Fiona Ebner <f.ebner@proxmox.com>
Reviewed-by: Jakob Klocker <j.klocker@proxmox.com>
Tested-by: Jakob Klocker <j.klocker@proxmox.com>
---
 vma-writer.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/vma-writer.c b/vma-writer.c
index 3f489092df..a85a1267a7 100644
--- a/vma-writer.c
+++ b/vma-writer.c
@@ -140,11 +140,15 @@ int vma_writer_add_config(VmaWriter *vmaw, const char *name, gpointer data,
     g_free(basename);
 
     if (!name_ptr) {
+        vma_writer_set_error(vmaw, "vma_writer_add_config: unable to allocate"
+                             " name_ptr");
         return -1;
     }
 
     uint32_t data_ptr = allocate_header_blob(vmaw, data, len);
     if (!data_ptr) {
+        vma_writer_set_error(vmaw, "vma_writer_add_config: unable to allocate"
+                             " data_ptr");
         return -1;
     }
 
